lib/Date/Utility.pm

```diff
@@ -840,6 +838,35 @@ sub months_ahead {
     return month_number_to_abbrev($new_month) . '-' . $new_year;
 }
 
+=head2 move_to_nth_dow
+
+Takes an integer as an ordinal and a day of week representation
+
+The following are all equivalent:
+C<move_to_nth_dow(3, 'Monday')>
+C<move_to_nth_dow(3, 'Mon')>
+C<move_to_nth_dow(3, 1)>
+
+Returning the 3rd Monday of the month represented by the object or
+C<undef> if it does not exist.
+
+An exception is thrown on improper day of week representations.
+
+=cut
+
+sub move_to_nth_dow {
+    my ($self, $nth, $dow_abb) = @_;
+
+    $dow_abb //= 'undef';    # For nicer error reporting below.
+
+    my $dow = $days_to_num{lc $dow_abb} // croak 'Invalid day of week. We got [' . $dow_abb . ']';
+
+    my $dow_first = ($self->day_of_month - $self->day_of_week) % 7;
+    my $dom = ($dow + 7 - $dow_first) % 7 + ($nth - 1) * 7 + 1;
+
+    return try { Date::Utility->new(join '-', $self->year, $self->month, $dom) };
+}
```

Comment:
Given the error above, I think this function needs better testing. The following piece is just an example. Ideally the `19` is also varied for all days of a month. Also 1000 days is just an arbitrary number.

```perl
    my $d = Date::Utility->new(Date::Utility::today->year . '-01-01 12:00');
    my @dow;
    my $M = '';
    for (0..1000) {
        unless ($M eq $d->year.$d->month) {
            $M = $d->year.$d->month;
            @dow = ();
        }
        $dow[$d->day_of_week]++;
        is +Date::Utility->new(join '-', $d->year, $d->month, 19)
                         ->move_to_nth_dow($dow[$d->day_of_week], $d->day_of_week)
                         ->date,
           $d->date,
           "$dow[$d->day_of_week]. ".$d->full_day_name." of ".$d->month_as_string.' '.$d->year.' is '.$d->date;

        $d = $d->plus_time_interval('1d');
    }
```